packages/pagination/src/Pagination.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useContext } from 'react';
+import PropTypes from 'prop-types';
+import isFunction from 'lodash.isfunction';
+import { useEffectAsync, useToggle } from '@availity/hooks';
+
+export const PaginationContext = React.createContext();
+
+export const usePagination = () => useContext(PaginationContext);
+
+// todos
+// Add another `useEffect` for only updating the items without calling the function if the `itemsPerPage` prop changes
+
+/**
+ * Pagination Provider that takes in an async function or items and provides a list of items
+ * @param {Function|Object} items - The Items or function to use for pagination
+ * @param {number} itemsPerPage - The items to show per page
+ */
+const Pagination = ({ items: theItems, itemsPerPage, children }) => {
+  const [currentPage, setPage] = useState(1);
+  const [pageData, setPageData] = useState({
+    total: theItems != null ? theItems.totalCount : 0,
```

Comment:
when `theItems` is a function, it will not be null and this will return `theItems.totalCount` which would be `undefined`. Not sure if that is intentional.

Reply:
No that was unintentional. Adding the `isFunction` check on there as well. 🐛 🔨